limit.go

```diff
@@ -114,74 +154,62 @@ func (l *BaseLimit) GetFDLimit() int {
 	return l.FD
 }
 
-func (l *BasicLimiter) GetSystemLimits() Limit {
-	return l.SystemLimits
+func (l *BaseLimit) GetMemoryLimit() int64 {
+	return l.Memory
 }
 
-func (l *BasicLimiter) GetTransientLimits() Limit {
-	return l.TransientLimits
+func (l *fixedLimiter) GetSystemLimits() Limit {
+	return &l.System
 }
 
-func (l *BasicLimiter) GetServiceLimits(svc string) Limit {
-	sl, ok := l.ServiceLimits[svc]
-	if !ok {
-		return l.DefaultServiceLimits
-	}
-	return sl
+func (l *fixedLimiter) GetTransientLimits() Limit {
+	return &l.Transient
 }
 
-func (l *BasicLimiter) GetServicePeerLimits(svc string) Limit {
-	pl, ok := l.ServicePeerLimits[svc]
+func (l *fixedLimiter) GetServiceLimits(svc string) Limit {
```

Comment:
Not for this PR, but I would like a method for Limiter that iterates over all limits so that we can expose this information to the use (via opencensus or something else).

Maybe something like `GetAllLimits(forEach func(scopeName string, Limit))`

Reply:
This would be good to log at the start as well since it can be a bit confusing what the final values are at the end of the day.